pkg/parse/updater.go

```diff
@@ -157,13 +157,19 @@ func (u *updater) Update(ctx context.Context, cache *cacheForCommit) status.Mult
 
 // update performs most of the work for `Update`, making it easier to
 // consistently prepend the conflict errors.
-func (u *updater) update(ctx context.Context, cache *cacheForCommit) status.MultiError {
+func (u *updater) update(ctx context.Context, cache *CacheForCommit) status.MultiError {
 	// Stop remediator workers.
 	// This prevents objects been updated in the wrong order (dependencies).
 	// Continue watching previously declared objects and updating the queue.
 	// Queued objects will be remediated when the workers are started again.
 	u.remediator.Pause()
 
+	// block and wait until the current remediation is done.
+	u.remediator.WaitRemediating()
```

Comment:
Could Pause do the waiting or does this need to be a separate call?

Reply:
Looks like this set of 3 calls is made in the same order in two places. Could it be reduced to one method?